compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -207,16 +212,18 @@ object MainGenericRunner {
       case ExecuteMode.Script =>
         val targetScript = Paths.get(settings.targetScript).toFile
         val targetJar = settings.targetScript.replaceAll("[.][^\\/]*$", "")+".jar"
-        val precompiledJar = Paths.get(targetJar).toFile
+        val precompiledJar = File(targetJar)
         val mainClass = if !precompiledJar.isFile then "" else Jar(targetJar).mainClass.getOrElse("")
-        val jarIsValid = mainClass.nonEmpty && precompiledJar.lastModified >= targetScript.lastModified
+        val jarIsValid = mainClass.nonEmpty && precompiledJar.lastModified >= targetScript.lastModified && settings.save
         if jarIsValid then
           // precompiledJar exists, is newer than targetScript, and manifest defines a mainClass
           sys.props("script.path") = targetScript.toPath.toAbsolutePath.normalize.toString
           val scalaClasspath = ClasspathFromClassloader(Thread.currentThread().getContextClassLoader).split(classpathSeparator)
           val newClasspath = (settings.classPath.flatMap(_.split(classpathSeparator).filter(_.nonEmpty)) ++ removeCompiler(scalaClasspath) :+ ".").map(File(_).toURI.toURL)
           if mainClass.nonEmpty then
-            ObjectRunner.runAndCatch(newClasspath :+ File(targetJar).toURI.toURL, mainClass, settings.scriptArgs)
+            ObjectRunner.runAndCatch(newClasspath :+ File(targetJar).toURI.toURL, mainClass, settings.scriptArgs).foreach { (t: Throwable) =>
+              t.printStackTrace()
+            }
```

Comment:
Isn't it returning `Unit` instead of `Option[Throwable]`?

Reply:
I guess it was my problem that I don't assign the throwable and then print it if something went wrong. I think it should be done that way, then we will reuse Exception from line 228

Reply:
So something like this?
```scala
          val res = if mainClass.nonEmpty then
            ObjectRunner.runAndCatch(newClasspath :+ File(targetJar).toURI.toURL, mainClass, settings.scriptArgs)
          else
            Some(IllegalArgumentException(s"No main class defined in manifest in jar: $precompiledJar"))
          errorFn("", res)
```

Reply:
Yes, something like that seems reasonable